packages/scm/src/browser/scm-tree-widget.tsx

```diff
@@ -605,7 +605,7 @@ export class ScmResourceComponent extends ScmElement<ScmResourceComponent.Props>
 
     protected readonly contextMenuPath = ScmTreeWidget.RESOURCE_CONTEXT_MENU;
     protected get contextMenuArgs(): any[] {
-        if (!this.props.model.selectedNodes.some(node => ScmFileChangeNode.is(node) && node.sourceUri === this.props.sourceUri)) {
+        if (!this.props.model.selectedNodes.some(node => ScmFileChangeNode.is(node) && node === this.props.treeNode)) {
```

Comment:
This is a bug fix. There can be more than one node here with the the same `sourceUri`. For example, in case of partially staged changes, there can be one node in the 'Staged Changes' group and another in the 'Changes' group, both having the same `sourceUri`.